Zadaca3/Animal.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Scanner;
+
+public abstract class Animal {
+    protected String name;
+    protected String food;
+    //protected int vaccinationCode;
+
+    protected abstract void animalSound();
+
+    //Animal();
+
+    final void setName() {
+        System.out.println("Unesi ime svoje zivotinje: ");
+        Scanner obj = new Scanner(System.in);
+        name = obj.nextLine();
+    }
+
+    protected final String getName() {
+        return name;
+    }
+
+    protected abstract void setFood();
+
+    protected final String getFood() {
+        return food;
+    }
+
+    // Regular method
+    final void sleep() {
+        System.out.println("Zzz");
+    }
+}
+
+// Subclass (inherit from Animal)
+class Dog extends Animal {
+    private String favoriteBone;
+    private boolean likesBigBones;
+
+    Dog() {
+        favoriteBone = new String();
```

Comment:
if you have default assignment which is not based on passing values to the constructor (aka there is no parameters in constructor), this assignment can happen directly on variables above, no need to do it in constructor